mitodl/mit-open
# Update readable_id and run for OCW courses

# What are the relevant tickets?
Closes #132 

# Description (What does it do?)
- Sets `readable_id` for OCW courses to be `<course_id>+<semester>_<year>`
- Sets `etl_source` value for OCW courses
- Adds a migration to update above values for existing OCW courses


# How can this be tested?
- Start with the **main** branch and try running the following (copy `OCW_*` values from RC to your .env file first):
  ```
  ./manage.py backpopulate_ocw_data  --course-name 21g-104-chinese-iv-regular-spring-2004
  ./manage.py backpopulate_ocw_data  --course-name 21g-104-chinese-iv-regular-spring-2006
  ./manage.py backpopulate_ocw_data  --course-name 21g-104-chinese-iv-regular-spring-2018
  ```
- You will end up with one LearningResource that has 3 runs.  This is actually not what is desired for OCW courses, because each run of a course has its own separate site url.
- Switch to this branch and restart containers.  A new migration should convert the readable_id for the LearningResource above to include the semester and year of one of the 3 runs, and the other 2 runs should be deleted.
- Run the 3 backpopulate commands again.  When finished, there should be 3 `LearningResource` objects with 1 run each and `readable_ids` of `21G.104+spring_2004`, `21G.104+spring_2006`, and `21G.104+spring_2018`
